src/Web/Telegram/API/Bot/API.hs

```diff
@@ -68,182 +73,179 @@ newtype Token = Token Text
 telegramBaseUrl :: BaseUrl
 telegramBaseUrl = BaseUrl Https "api.telegram.org" 443 ""
 
--- | Type for token
-type TelegramToken = Capture ":token" Token
-
 -- | Telegram Bot API
 type TelegramBotAPI =
```

Comment:
This ClientM is already provided by Servant. You just need to expose it.

But if you want (I do want) to hide token in such a wrapper, then you heed either (1) wrap ClientM in another ReaderT, or (2) treat token as a part of base URL internally and provide convenient runClient runner for this monad.